hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/upgrade/AbstractLayoutVersionManager.java

```diff
@@ -129,20 +129,18 @@ public void finalized(T layoutFeature) {
           LOG.info("Finalization is complete.");
         }
       } else {
-        String msgStart = "";
+        String versionMsg = "Software layout version: " + softwareLayoutVersion
+            + " Metadata layout version: " + metadataLayoutVersion
+            + " Feature Layout version: " + layoutFeature.layoutVersion();
+
         if (layoutFeature.layoutVersion() <= metadataLayoutVersion) {
-          msgStart = "Finalize attempt on a layoutFeature which has already "
-              + "been finalized.";
+          LOG.warn("Finalize attempt on a layoutFeature which has already "
+              + "been finalized. " + versionMsg);
```

Comment:
Since this can happen normally maybe it should be an info message, and explain that this is expected when Ratis logs are replayed?

Reply:
Makes sense. Updated the PR accordingly.